arch/arm/src/samv7/sam_usbdevhs.c

```diff
@@ -458,7 +458,7 @@ static void   sam_putreg(uint32_t regval, uintptr_t regaddr);
 static void   sam_dumpep(struct sam_usbdev_s *priv, int epno);
 #else
 static inline uint32_t sam_getreg(uintptr_t regaddr);
-static inline void sam_putreg(uint32_t regval, uintptr_t regaddr);
```

Comment:
But, other similar functions(e.g. sam_getreg) can pass the build even it use uintptr_t. My question is that why only sam_putreg fail to compile. Could you point me which line fail to compile and the detailed error message?

Reply:
My bad, `sam_getreg()` also fail to compile without changes. I just forget to commit:

```
CC:  chip/sam_usbdevhs.c
chip/sam_usbdevhs.c:830:24: error: conflicting types for 'sam_getreg'
 static inline uint32_t sam_getreg(uint32_t regaddr)
                        ^~~~~~~~~~
chip/sam_usbdevhs.c:460:24: note: previous declaration of 'sam_getreg' was here
 static inline uint32_t sam_getreg(uintptr_t regaddr);
                        ^~~~~~~~~~
chip/sam_usbdevhs.c:856:20: error: conflicting types for 'sam_putreg'
 static inline void sam_putreg(uint32_t regval, uint32_t regaddr)
                    ^~~~~~~~~~
chip/sam_usbdevhs.c:461:20: note: previous declaration of 'sam_putreg' was here
 static inline void sam_putreg(uint32_t regval, uintptr_t regaddr);
```

Reply:
Ok, the better fix is correct the prototype at line 856:
```
static inline void sam_putreg(uint32_t regval, uintptr_t regaddr)
```